service/resource/resourcegroup/resource.go

```diff
@@ -0,0 +1,289 @@
+// deleted.
+func (r *Resource) GetDeleteState(obj, currentState, desiredState interface{}) (interface{}, error) {
+	currentResourceGroup, err := toGroup(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredResourceGroup, err := toGroup(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var resourceGroupToDelete *Group
+	if currentResourceGroup != nil && currentResourceGroup.Name != "" {
+		resourceGroupToDelete = desiredResourceGroup
+	}
+
+	return resourceGroupToDelete, nil
+}
+
+// GetUpdateState returns an empty group for the create, delete and update
+// states because resource groups are not updated.
+func (r *Resource) GetUpdateState(obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error) {
+	return &Group{}, &Group{}, &Group{}, nil
+}
+
+// Name returns the resource name.
+func (r *Resource) Name() string {
+	return Name
+}
+
+// ProcessCreateState creates the resource group via the Azure API.
+func (r *Resource) ProcessCreateState(obj, createState interface{}) error {
+	customObject, err := toCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	resourceGroupToCreate, err := toGroup(createState)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if resourceGroupToCreate != nil && resourceGroupToCreate.Name != "" {
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "creating the resource group in the Azure API")
+
+		groupClient, err := r.getGroupsClient()
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		resourceGroup := azureresource.Group{
+			Name:      to.StringPtr(resourceGroupToCreate.Name),
+			Location:  to.StringPtr(resourceGroupToCreate.Location),
+			ManagedBy: to.StringPtr(managedBy),
+			Tags:      to.StringMapPtr(resourceGroupToCreate.Tags),
+		}
+		_, err = groupClient.CreateOrUpdate(resourceGroupToCreate.Name, resourceGroup)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "created the resource group in the Azure API")
+	}
+
+	return nil
+}
+
+// ProcessDeleteState deletes the resource group via the Azure API.
+func (r *Resource) ProcessDeleteState(obj, deleteState interface{}) error {
+	customObject, err := toCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	resourceGroupToDelete, err := toGroup(deleteState)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if resourceGroupToDelete != nil && resourceGroupToDelete.Name != "" {
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "deleting the resource group in the Azure API")
+
+		groupsClient, err := r.getGroupsClient()
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		// Delete the resource group which also deletes all resources it
+		// contains. We wait for the error channel while the deletion happens.
+		_, errchan := groupsClient.Delete(resourceGroupToDelete.Name, nil)
+		select {
```

Comment:
Here we wait for the error channel or time out after 5 mins. Deleting the resource group is a long running operation if it has a lot of resources.